pkg/parser/extended_syntax_parser_test.go

```diff
@@ -240,18 +248,34 @@ func TestExtendedSyntaxParseSelectStarAndPrint(t *testing.T) {
 	a.Equal("SELECT *, b\nFROM a\nLIMIT 10", r.StandardSelect.String())
 }
 
-func TestExtendedSyntaxParseStandardDropTable(t *testing.T) {
+func TestExtendedSyntaxParseNonSelectStmt(t *testing.T) {
 	a := assert.New(t)
-	_, e := parseSQLFlowStmt(`DROP TABLE TO PREDICT`)
-	a.Error(e)
-	// Note: currently, our parser doesn't accept anything statements other than SELECT.
-	// It will support parsing any SQL statements and even dialects in the future.
+	{
+		r, idx, e := parseSQLFlowStmt(`DROP TABLE TO PREDICT`)
+		a.Nil(r)
+		a.Equal(0, idx)
+		a.Error(e)
+	}
+	{
+		r, idx, e := parseSQLFlowStmt(`   DROP TABLE TO PREDICT`)
+		a.Nil(r)
+		a.Equal(3, idx) // right before DROP as there was an error.
+		a.Error(e)
+	}
```

Comment:
Maybe add more tests on statements like
1. Missing WITH clause `select ... to train dnn label class ...`
1. Missing semicolon in the with clause `select ... to train dnn with a = b c = d...`.
1. Addition semicolon in the with clause `select ... to train dnn with a = b, c=d, column ...`

Users often make these mistakes. And the error message from the parser can be really helpful.